lib/interactive/control-router.js

```diff
@@ -19,45 +19,37 @@ function controlRouter(mouseevent, mixerControls, mixerControl, gameJson, inputE
       username = participant.username;
     }
 
-    // Log input event.
-    //console.log(inputEvent);
-    //console.log(participant);
-    //console.log(control);
+    // Create request wrapper (instead of having to pass in a ton of args)
+    var processEffectsRequest = {
+        effects: effects,
+        firebot: firebot,
+        participant: participant,
+        control: control,
+        isManual: false
+    } 
 
     // Check to see if this is a mouse down or mouse up event.
     if(mouseevent == "mousedown"){
         // Mouse Down event called.
 
-        // Cooldown Button or Buttons
-        cooldown.router(mixerControls, mixerControl, firebot, control, function(response){
-                  
-          // Throw this information into the moderation panel.
-          renderWindow.webContents.send('eventlog', {username: participant.username, event: "pressed the "+controlID+" button."});
-
-          // Charge sparks for the button that was pressed.
-          if (inputEvent.transactionID) {
-              // This will charge the user.
-              mixerInteractive.sparkTransaction(inputEvent.transactionID);
-          };
-          
-          // Create request wrapper (instead of having to pass in a ton of args)
-          var processEffectsRequest = {
-            effects: effects,
-            firebot: firebot,
-            participant: participant,
-            control: control,
-            isManual: false
-          }        
-          
-          // Run the effects
-          effectRunner.processEffects(processEffectsRequest)
-            .then(function() {
-              // This is called after the effects are done running. 
-            });
-
-        }) // end cooldown function
+        // Make sure cooldown is processed.
+        cooldown.router(mixerControls, mixerControl, firebot, control)
+        .then((res) => autoPlay(processEffectsRequest))
```

Comment:
@Firebottle This should work fine, but I am unclear of intended behavior. Do you want to deduct sparks after all the button effects have finished? The way you have a .then after calling the autoPlay function seems to imply that, however this setup doesn't technically wait for the effects to finish (if there is delays, anyway). 

If you don't wish to wait for effects to finish (What makes sense to me, deduction should happen immediately in my eyes), then you just could do:
```javascript
cooldown.router(mixerControls, mixerControl, firebot, control)
    .then((res) => { 
         // Start effect execution
         autoPlay(processEffectsRequest);

         // Charge sparks for the button that was pressed.
         if (inputEvent.transactionID) {
             // This will charge the user.
             mixerInteractive.sparkTransaction(inputEvent.transactionID);
         };
    })
```
That will make it more clear that you don't necessarily expect autoPlay to finish and so this removes the extra 'then'.

If you do want to wait for effects to finish, however, then you will need autoPlay to return the effect-runner promise.
```javascript
function autoPlay(processEffectsRequest){
     // Run the effects
     return effectRunner.processEffects(processEffectsRequest);
 }
```
then you can do :
```javascript
cooldown.router(mixerControls, mixerControl, firebot, control)
    .then((res) => { 
         // Start effect execution
         return autoPlay(processEffectsRequest);
    })
    .then(() => { 
         // Charge sparks for the button that was pressed.
         if (inputEvent.transactionID) {
             // This will charge the user.
             mixerInteractive.sparkTransaction(inputEvent.transactionID);
         };
    })
```
Ultimately this is just a style change, not a functional change, so at the end of the day it's personal preference. Whatever you prefer! Looks like it will work great as is.